### PR TITLE
Modify search

### PR DESCRIPTION
**Pull request by skwiens**
- categories method in product model modified to only return categories associated with active products
- new method in merchants method to return a unique array of merchants with active products
- index product page changed so that user can only filter by active methods and active merchants

### CODE REVIEW

File: app/models/merchant.rb [hunk 1]
```diff
@@ -13,4 +13,10 @@ def join_orderitems(status)
       return Orderitem.joins('join products on orderitems.product_id = products.id').where("products.merchant_id = #{self.id}").joins('LEFT JOIN orders ON orderitems.order_id = orders.id').where("orders.status = '#{status}' AND orders.status IS DISTINCT FROM 'pending'")
     end
   end
+
+  def self.active_merchants
+    merchants = Product.where(status: "active").pluck(:merchant_id).uniq
```
Comment (vsawchuk) [inline app/models/merchant.rb:18]: nice!

File: test/models/product_test.rb [hunk 2]
```diff
@@ -202,14 +203,14 @@
   end
 
   describe "self.retired_only" do
```
Comment (vsawchuk) [inline test/models/product_test.rb]: if we're removing the method, let's remove the tests too

**Review by vsawchuk — CHANGES_REQUESTED** [review]
Once I edit a product to be retired, that works. But when I go back to the edit page, it defaults to an active status. We should use the current status to set the status on the edit form.

**Comment by skwiens** [comment]
Thanks! Made the changes...

**Review by vsawchuk — APPROVED** [review]

**Pull request merged**